lib/awful/widget/watch.lua

```diff
@@ -79,7 +80,7 @@ function watch.new(command, timeout, callback, base_widget)
     end)
     t:start()
     t:emit_signal("timeout")
-    return base_widget
+    return { widget = base_widget, timer = t }
```

Comment:
> If this is a good idea is for others to decide...

Starting from `vicious`, every widget library enables you to control your widget timers. I'll give you an example: if I have a network resources fetcher with a timeout of a couple of seconds and I don't want it to watch when I am away from my machine, I could disable it and spare resources. Now imagine if one has multiple watchers.

If you don't find it useful, I do. 

Plus, returning multiple arguments, if you don't want the timer, you can just skip it.